hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/has/isauthorizedraw/IsAuthorizedRawCall.java

```diff
@@ -0,0 +1,266 @@
+        } else key = Optional.empty();
+
+        // Key must be simple (for isAuthorizedRaw)
+        if (key.isPresent()) {
+            final Key ky = key.get();
+            final boolean keyIsSimple = !ky.hasKeyList() && !ky.hasThresholdKey();
+            if (!keyIsSimple) return bail.apply(FAIL_INVALID /* should be: "account key must be simple" */);
+        }
+
+        // Key must match signature type
+        if (key.isPresent()) {
+            if (!switch (signatureType) {
+                case ED -> key.get().hasEd25519();
+                case EC -> false;
+                default -> false;
+            }) return bail.apply(INVALID_SIGNATURE_TYPE_MISMATCHING_KEY);
+        }
+
+        // Finally: Do the signature validation we came here for!
+        final boolean authorized =
+                switch (signatureType) {
+                    case EC -> validateEcSignature(address, frame);
+                    case ED -> validateEdSignature(account, key.get());
+                    default -> false;
+                };
+
+        final var result = gasOnly(
+                successResult(encodedAuthorizationOutput(authorized), HARDCODED_GAS_REQUIREMENT_GAS), SUCCESS, true);
+        return result;
+    }
+
+    /** Validate EVM signature - EC key - via the ECRECOVER precompile */
+    public boolean validateEcSignature(@NonNull final Address address, @NonNull final MessageFrame frame) {
+        requireNonNull(address, "address");
+        requireNonNull(frame, "frame");
+
+        // Call the ECRECOVER precompile directly (meaning, not by executing EVM bytecode, just by
+        // using the class provided by Besu).
+
+        final var input = formatEcrecoverInput(messageHash, signature);
+        if (input.isEmpty()) return false;
+        final var ecrecoverResult = ecPrecompile.computePrecompile(Bytes.wrap(input.get()), frame);
+
+        // ECRECOVER's output always has status SUCCESS but the result Bytes are either empty or the recovered address.
+        if (ecrecoverResult.getOutput().isEmpty()) return false;
+        final var recoveredAddress =
+                ecrecoverResult.getOutput().slice(12).toArray(); // LAST 20 bytes are where the EVM address is
+
+        // ECRECOVER produced an address:  Must match our account's alias address
+        final var givenAddress = explicitFromHeadlong(address);
+        final var addressesMatch = 0 == Arrays.compare(recoveredAddress, givenAddress);
+
+        return addressesMatch;
+    }
+
+    /** Validate (native Hedera) ED25519 signature */
+    public boolean validateEdSignature(@NonNull final Account account, @NonNull final Key key) {
+        requireNonNull(account, "account");
+        requireNonNull(key, "key");
+
+        // Use of `com.swirlds.common.crypto.CryptographyHolder` straight from the Platform is deprecated:
+        // FUTURE: Get the `Cryptography` engine from the services app via the context (needs to be invented)
+
+        return CryptographyHolder.get()
```

Comment:
Validation if the account matches the one from the decoded signature is not permed, basically the `account` input is ignored. 
Since in Hedera multiple accounts can have the same key. This could lead to bug where if any address that has the same key and is passed as a parameter isAuthorizedRaw would also return true!

With @mustafauzunn we've added a case scenario that covers that, and currently it fails, due to the missing validation.
Check out this test case: `callWithDifferentAccountsWithSameKey`